smashed/mappers/tokenize.py

```diff
@@ -59,16 +61,37 @@ def __init__(
         if return_length:
             output_fields.append(f'{self.prefix}length')
 
+        if 'is_split_into_words' in tk_kwargs and return_word_ids:
+            output_fields.append(f'{self.prefix}word_ids')
+            if return_words:
+                output_fields.append(f'{self.prefix}words')
+
         self.tokenize_kwargs = tk_kwargs
         self.input_fields = input_fields
         self.output_fields = output_fields
 
     def transform(self, data: TransformElementType) -> TransformElementType:
         batch_encoding = self.tokenizer(data[self.input_fields[0]],
                                         **self.tokenize_kwargs)
-        batch_encoding = {f'{self.prefix}{k}': v
-                          for k, v in batch_encoding.items()}
-        return dict(batch_encoding)
+        batch_encoding_as_dict = {f'{self.prefix}{k}': v
```

Comment:
I would do this as last step: use `BatchEncoding` object as normal, but the, before returning, call a function that adds the prefix + casts

Reply:
done